multicloser/multicloser.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package multicloser provides a convenient way to join multiple "close"
+// functions together so they can be called together. This is especially useful
+// to group multiple cleanup function calls and return it as a single "closer"
+// to be called later.
+package multicloser
+
+import (
+	"errors"
+)
+
+// Func is the type signature for a closing function. It accepts a function that
+// returns an error or a void function.
+type Func interface {
+	func() error | func()
```

Comment:
You **cannot** make these concrete types or the type system freaks out. This is the situation I was running into the first time I tried.